File: Python/03. Code Related to Arrays/70. Finding Equilibrium index of an array.py
size = int(input("Enter the length of an Array:"))
array = []
for i in range(size):
    element = int(input(f"Enter {i+1} element :"))
    array.append(element)

total = sum(array)
leftSideAdd = 0
found = False

for i,number in enumerate(array):
    rightSideAdd = total - leftSideAdd - number

    if( leftSideAdd == rightSideAdd ):
        print("The index", i, "is the Equilibrium Index")
        break

    leftSideAdd += number
else:
    print("no equilibrium found")

File: Java/03. Code Related to Arrays/70. Finding Equilibrium index of an array.java
import java.util.*;

class Main {
    public static void main(String [] args) {
        Scanner userInput = new Scanner(System.in);

        System.out.print("Enter the number of elements of the Array : ");
        int size = userInput.nextInt();
        int[] numbers = new int[size];
        int total = 0;
        boolean find = false;

        for (int i = 0; i < size; i++) {
            System.out.print("Element " + (i + 1) + ": ");
            numbers[i] = userInput.nextInt();
            total += numbers[i];
        }
        int leftSideAdd = 0;

        for (int i = 0; i < size; i++) {
            int number = numbers[i];
            int rightSideAdd = total - leftSideAdd - number;

            if(leftSideAdd == rightSideAdd) {
                System.out.println("The index "+i+" is the Equilibrium Index");
                find = true;
                break;
            }
            leftSideAdd += number;
        }
        if(!find){
            System.out.println("no equilibrium found!");
        }
        userInput.close();
    }
}